src/js/media-fullscreen-button.js

```diff
@@ -20,30 +20,69 @@ class MediaFullscreenButton extends MediaChromeButton {
     super();
 
     this.icon = enterFullscreenIcon;
+  }
 
-    this.addEventListener("click", e => {
-      const media = this.media;
-
-      if (this.mediaChrome == document.fullscreenElement) {
-        document.exitFullscreen();
-      } else {
-        if (document.pictureInPictureElement) {
-          // Should be async
-          document.exitPictureInPicture();
-        }
+  onClick() {
+    const api = this.getFullscreenApi(this.mediaChrome);
 
-        this.mediaChrome.requestFullscreen();
+    if (this.mediaChrome == document[api.element]) {
+      api.exit();
+    } else {
+      if (document.pictureInPictureElement) {
+        // Should be async
+        document.exitPictureInPicture();
       }
-    });
 
-    document.addEventListener("fullscreenchange", () => {
-      if (this.mediaChrome == document.fullscreenElement) {
+      api.enter();
+    }
+  }
+
+  mediaSetCallback() {
+    const api = this.getFullscreenApi(this.mediaChrome);
+    
+    document.addEventListener(api.event, () => {
+      if (this.mediaChrome == document[api.element]) {
```

Comment:
The `getFullscreenApi` method needs a reference to the `media-chrome` element, so the fullscreen event listener has been moved to the `mediaSetCallback`